source/parse.h

```diff
@@ -5446,6 +5482,21 @@ class parser
             return {};
         }
 
+        if (
+            n->is_function()
+            && n->initializer
+            && !done() && curr().type() == lexeme::Semicolon
+            )
+        {
+            if (n->initializer->is_compound() && n->has_name()) {
+                error("a compound function body shall not end with a semicolon");
```

Comment:
"a braced function body may not be followed by a semicolon (empty statements are not allowed)"